src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -173,6 +225,36 @@ static bool xccdf_policy_filter_selected(void *item, void *policy)
 		(xccdf_select_get_selected((struct xccdf_select *) item)));
 }
 
+static xccdf_role_t _get_final_role(const struct xccdf_rule *rule, const struct xccdf_refine_rule_internal* r_rule){
+	if (r_rule == NULL){
+		return xccdf_rule_get_role(rule);
+	} else {
+		return r_rule->role;
+	}
+}
+
+static inline bool _weight_defined(xccdf_numeric weight){
```

Comment:
There is already function xccdf_refine_rule_weight_defined() in profile.c, but do two things (access to structure + check if weight is defined); In my opinion, It should be merged somehow in next iteration.